### PR TITLE
Serious typeo in sched_switch code (& and ==)

### PR DESCRIPTION
**Pull request by coodie**
Shouldn't this be "==" instead of "&" ?

By the way, since me and @rafalcieslak managed to find some bugs in @cahirwpz implementation, wouldn't it be good idea to make pull requests so everyone can do code review?

### CODE REVIEW

File: sys/sched.c [hunk 1]
```diff
@@ -54,7 +54,7 @@ void sched_switch(thread_t *newtd) {
 
   td->td_flags &= ~(TDF_SLICEEND | TDF_NEEDSWITCH);
 
-  if (td->td_state & TDS_RUNNING)
+  if (td->td_state && TDS_RUNNING)
```
Comment (cahirwpz) [inline sys/sched.c]: equivalent to `(td->td_state != 0) && (TDS_RUNNING != 0)`

File: sys/sched.c [hunk 2]
```diff
@@ -54,7 +54,7 @@ void sched_switch(thread_t *newtd) {
 
   td->td_flags &= ~(TDF_SLICEEND | TDF_NEEDSWITCH);
 
-  if (td->td_state & TDS_RUNNING)
+  if (td->td_state == TDS_RUNNING)
```
Comment (cahirwpz) [inline sys/sched.c]: Right! Nice find!

**Comment by cahirwpz** [comment]
I'm a human and I make mistakes – I'll ask both of you for review when my changes are non-trivial.

**Comment by rafalcieslak** [comment]
You just saved me a lot of debugging!

But I doubt I could spot this if I were to review this code.

**Comment by cahirwpz** [comment]
*"Given enough eyeballs, all bugs are shallow."* – Linus Torvalds ;-)

**Pull request merged**